tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -606,53 +607,85 @@ func (a *AwsProvider) createWindowsWorkerSg(infraID string, vpc *ec2.Vpc) (strin
 	return *sg.GroupId, nil
 }
 
-// authorizeSgIngress attaches inbound rules for RDP from user's external IP address and all traffic within the VPC
-// on newly created security group for Windows instance. The function returns error if failed.
-func (a *AwsProvider) authorizeSgIngress(sgID string, vpc *ec2.Vpc) error {
-	myIP, err := GetMyIp()
-	if err != nil {
-		return err
+//GetMissingPorts returns a slice of missing ports from the rules present in local IP
+func GetMissingPorts(myIP string, rules []*ec2.IpPermission) []int64 {
+	//var hasWinRM, hasSSH, hasRDP bool
+	var ports []int64
+	mapPorts := map[int64]bool{}
+	// Loop through the rules, then loop through the ips to get a map of ports
+	for _, rule := range rules {
+		for _, ips := range rule.IpRanges {
+			if *ips.CidrIp == myIP+"/32" {
+				mapPorts[*rule.FromPort] = true
+			}
+		}
 	}
 
-	_, err = a.EC2.AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
-		GroupId: aws.String(sgID),
-		IpPermissions: []*ec2.IpPermission{
+	//Add missing ports to a slice ports
+	if !mapPorts[WINRM_PORT] {
+		ports = append(ports, WINRM_PORT)
+	}
+	if !mapPorts[sshPort] {
+		ports = append(ports, sshPort)
+	}
+	if !mapPorts[rdpPort] {
+		ports = append(ports, rdpPort)
+	}
+	return ports
+}
+
+// GetMissingRules gets a slice of rules to serve an input for IpPermissions for ec2.AuthorizeSecurityGroupIngressInput
+// method used in SetSgIngress function. The function returns error if failed.
+func (a *AwsProvider) GetMissingRules(myIP string, rules []*ec2.IpPermission, vpcCidr string) ([]*ec2.IpPermission, error) {
+	// slice of missing ports in the sg for myIP
+	var hasAllPorts bool
+
+	// Get the missing ports
+	ports := GetMissingPorts(myIP, rules)
+
+	// empty slice to add missing rules in local IP
```

Comment:
I don't understand this comment. Do you mean `rules with local IP`?

Reply:
comment seems unclear, meant "declaring an empty slice to add the missing rules in sg with local IP", will update it. Thanks for the review.